its/tester.go

```diff
@@ -0,0 +1,218 @@
+	if count != len(dmt.Lines) {
+		dmt.fail("%d lines but got %d", count, len(dmt.Lines))
+	}
+	return dmt
+}
+
+func (dmt *dockerMachineTest) ShouldContainLine(index int, text string) IntegrationTest {
+	if index >= len(dmt.Lines) {
+		dmt.Failed = true
+		dmt.fail("at least %d lines\nGot %d", index+1, len(dmt.Lines))
+	} else if !strings.Contains(dmt.Lines[index], text) {
+		dmt.Failed = true
+		dmt.fail("line %d to contain '%s'\nGot '%s'", index, text, dmt.Lines[index])
+	}
+	return dmt
+}
+
+func (dmt *dockerMachineTest) ShouldEqualLine(index int, text string) IntegrationTest {
+	if index >= len(dmt.Lines) {
+		dmt.Failed = true
+		dmt.fail("at least %d lines\nGot %d", index+1, len(dmt.Lines))
+	} else if text != dmt.Lines[index] {
+		dmt.Failed = true
+		dmt.fail("line %d to be '%s'\nGot '%s'", index, text, dmt.Lines[index])
+	}
+	return dmt
+}
+
+func (dmt *dockerMachineTest) ShouldSucceed() IntegrationTest {
+	if dmt.Err != nil {
+		dmt.Failed = true
+		dmt.fail("to succeed\nFailed with %s", dmt.Err)
+	}
+	return dmt
+}
+
+func (dmt *dockerMachineTest) ShouldSucceedWith(text string) IntegrationTest {
+	if dmt.Err != nil {
+		dmt.Failed = true
+		dmt.fail("to succeed\nFailed with %s", dmt.Err)
+	} else if !strings.Contains(dmt.RawOutput, text) {
+		dmt.Failed = true
+		dmt.fail("output to contain '%s'\nGot '%s'", text, dmt.RawOutput)
+	}
+	return dmt
+}
+
+func (dmt *dockerMachineTest) ShouldFail() IntegrationTest {
+	if dmt.Err == nil {
+		dmt.Failed = true
+		dmt.fail("to fail\nGot success")
+	}
+	return dmt
+}
+
+func (dmt *dockerMachineTest) ShouldFailWith(text string) IntegrationTest {
+	if dmt.Err == nil {
+		dmt.Failed = true
+		dmt.fail("to fail\nGot success")
+	} else if !strings.Contains(dmt.RawOutput, text) {
+		dmt.Failed = true
+		dmt.fail("output to contain '%s'\nGot '%s'", text, dmt.RawOutput)
+	}
+	return dmt
+}
+
+func (dmt *dockerMachineTest) testedBinary() string {
+	var path string
+	if runtime.GOOS == "windows" {
+		path = "..\\bin\\docker-machine.exe"
+	} else {
+		path = "../bin/docker-machine"
+	}
+
+	_, err := os.Stat(path)
+	if err != nil {
+		dmt.t.Fatalf("%s binary not found", path)
+		return ""
+	}
+
+	return path
+}
+
+func (dmt *dockerMachineTest) replaceMachinePath(commandLine string) string {
+	return strings.Replace(commandLine, "machine", dmt.testedBinary(), -1)
+}
+
+func (dmt *dockerMachineTest) replaceDriver(commandLine string) string {
+	return strings.Replace(commandLine, "$DRIVER", dmt.DriverName(), -1)
+}
+
+func parseFields(commandLine string) []string {
+	fields := regexpCommandLine.FindAllString(commandLine, -1)
+
+	for i := range fields {
+		if len(fields[i]) > 2 && strings.HasPrefix(fields[i], "'") && strings.HasSuffix(fields[i], "'") {
+			fields[i] = fields[i][1 : len(fields[i])-1]
+		}
+	}
+
+	return fields
+}
+
+func (dmt *dockerMachineTest) Cmd(commandLine string) IntegrationTest {
+	if strings.HasPrefix(commandLine, "machine ") {
```

Comment:
Wouldn't it be pretty reasonable to assume that the invoked command is always `docker-machine`?  Why not remove the initial `machine` from the calls and skip this bit? e.g.:

```go
test.Cmd("ls -q --filter name=^t.*e[45]").ShouldSucceed()
```

Could even change the function name to `Machine` to be more clear:

```go
test.Machine("ls -q").ShouldSucceed().ShouldContainLines(0)
```